dali/sasha/sadbghk.cpp

```diff
@@ -0,0 +1,224 @@
+            if (iDirFile.isDirectory() == fileBool::foundYes && *dirPath != '.')
+            {
+                // Ensure directory name only
+                StringBuffer dirNameOnly;
+                String dirNameStr(dirPath);
+                int fwdSlashIndex = dirNameStr.lastIndexOf('/');
+                if (fwdSlashIndex > -1)
+                {
+                    String *tmpStr = dirNameStr.substring(fwdSlashIndex + 1);
+                    dirNameOnly.append(*tmpStr);
+                    delete tmpStr;
+                }
+                else
+                {
+                    dirNameOnly.append(dirNameStr);
+                }
+
+                if (isPostMortemDirPath(dirNameOnly) && isExpiredDirPath(dirNameOnly, defaultExpireDays))
+                {
+                    recursiveRemoveDirectory(dirPath);
+                    PROGLOG(LOGDBGHK "Deleted %s", dirPath);
+                }
+            }
+        }
+        pDirIter.clear();
+
+        PROGLOG(LOGDBGHK "%s", stopped ? "Stopped" : "Done");
+    }
+
+    int run()
+    {
+        unsigned interval = props->getPropInt("@interval", DEFAULT_EXPIRY_INTERVAL);
+        if (!interval)
+            stopped = true;
+        PROGLOG(LOGDBGHK "min interval = %d hr", interval);
+        unsigned initinterval = (interval - 1) / 2; // wait a bit til dali has started
+        CSashaSchedule schedule;
+        if (interval)
+            schedule.init(props, interval, initinterval);
+        initinterval *= 60 * 60 * 1000; // ms
+        unsigned started = msTick();
+        while (!stopped)
+        {
+            stopsem.wait(1000 * 60);
+            if (stopped)
+                break;
+            if (!interval || ((started != (unsigned)-1) && (msTick() - started < initinterval)))
+                continue;
+            started = (unsigned)-1;
+            if (!schedule.ready())
+                continue;
+            try
+            {
+                runDebugHousekeeping();
+            }
+            catch (IException *e)
+            {
+                StringBuffer s;
+                EXCLOG(e, LOGDBGHK);
+                e->Release();
+            }
+        }
+        PROGLOG(LOGDBGHK "Exit");
+        return 0;
+    }
+
+private:
+    bool isPostMortemDirPath(const StringBuffer &dirName)
+    {
+        // Expecting a directory name like "W20250225-101112"
+        RegExpr postMortemDirRegEx("^W[0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]-[0-9][0-9][0-9][0-9][0-9][0-9]$");
+        if (postMortemDirRegEx.find(dirName.str()))
+        {
+            PROGLOG(LOGDBGHK "Post-mortem dir: %s", dirName.str());
+
+            return true;
+        }
+        else
+        {
+            PROGLOG(LOGDBGHK "Non post-mortem dir: %s", dirName.str());
+
+            return false;
+        }
+    }
+
+    bool isExpiredDirPath(StringBuffer &dirName, const unsigned &defaultExpireDays)
```

Comment:
see other comments, should use iFile and iFile.getTime to look at modified-time.